src/main/webapp/app/admin/metrics/blocks/metrics-modal-threads/metrics-modal-threads.component.ts

```diff
@@ -50,8 +71,28 @@ export class MetricsModalThreadsComponent implements OnInit {
         }
     }
 
-    getThreads(): Thread[] {
-        return this.threads?.filter((thread) => !this.threadStateFilter || thread.lockName?.includes(this.threadStateFilter)) ?? [];
+    private isMatchingTextFilter(thread: Thread): boolean {
+        if (this.threadFilter == undefined) {
+            return true;
+        }
+
+        // Filter the threads only on the visible attributes
+        const filteredAttributes = ['threadName', 'threadId', 'blockedTime', 'blockedCount', 'waitedTime', 'waitedCount', 'lockName'];
+        return Object.keys(thread)
+            .filter((key) => filteredAttributes.includes(key))
+            .some((key) => thread[key]?.toString().includes(this.threadFilter!));
+    }
+
+    private isMatchingSelectedThreadState(thread: Thread): boolean {
+        if (this.selectedThreadState == undefined) {
+            return true;
+        }
+
+        return thread.threadState === this.selectedThreadState!;
+    }
+
+    refreshTheFilter() {
```

Comment:
I'd call it `refreshFilteredThreads`, we do not use articles in method names I think and as the property `threadFilter` also exists, this naming is misleading - it's nothing to do with `threadFilter` but I would assume so when reading the name.